model/service_instance.go

```diff
@@ -63,6 +64,7 @@ func (inst *ServiceInstance) Stop() {
 }
 
 type singletonTracerProvider struct {
```

Comment:
similar to other comments, if we just pass around `TracerProvider` we won't need this struct.